extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/PgPoolCreator.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.reactive.pg.client;
+
+import io.vertx.core.Vertx;
+import io.vertx.pgclient.PgConnectOptions;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.PoolOptions;
+
+/**
+ * This interface is an integration point that allows users to use {@link Vertx}, {@link PoolOptions} and
+ * {@link PgConnectOptions}
+ * configured automatically by Quarkus, but provide on a custom strategy for creating the final {@link PgPool}, instead of
+ * letting Quarkus create it (by calling {@link PgPool#pool(Vertx, PgConnectOptions, PoolOptions)}).
+ *
+ * Implementations of this class are meant to be used as CDI beans - if a single implementation marked as a CDI bean,
+ * it will be picked up by Quarkus and used to create the pool.
+ */
+public interface PgPoolCreator {
+
+    PgPool create(Input input);
+
+    interface Input {
+
+        Vertx vertx();
+
+        PoolOptions poolOptions();
+
+        PgConnectOptions pgConnectOptions();
+
+        /**
+         * The datasource name for which the pool is being created
+         */
+        String dataSourceName();
```

Comment:
I am not 100% sure about this... Maybe this should not be part of the API but we should instead use CDI qualifiers. I think this approach would however be simpler

Reply:
The qualifier approach sounds more consistent with the rest of the platform, but I'm less knowledgeable than you are.

Perhaps @gsmet will have an opinion on this.

Reply:
Yeah, I think I would go with a qualifier approach. That would make it easy to have specific creators for each datasource.

Reply:
I'll have a look this week

Reply:
@geoand did you make progress on this?

Reply:
Unfortunately I have not looked at it.

I will be traveling this week, so I only have a few hours in which I'll try and update this PR :)